data/src/main/kotlin/com/vultisig/wallet/data/api/models/cosmos/CosmosBalance.kt

```diff
@@ -8,5 +9,34 @@ data class CosmosBalance(
     @SerialName("denom")
     val denom: String,
     @SerialName("amount")
-    val amount: String
-)
\ No newline at end of file
+    val amount: String,
+) {
+    fun hasValidDenom(
+        coin: Coin,
+    ) = denom.isValidDenom(coin.ticker)
+            || denom.isValidKujiraFactoryDenom(coin.ticker)
+            || denom.equals(
+        coin.contractAddress,
+        ignoreCase = true
+    )
+    
+    private fun String.isValidDenom(ticker: String) =
+        equals(
+            ("u$ticker"),
+            ignoreCase = true
+        ) || equals(
+            ("a$ticker"),
+            ignoreCase = true
+        )
+
+    private fun String.isValidKujiraFactoryDenom(ticker: String): Boolean {
+        val value = removePrefix("factory/").substringAfter('/')
```

Comment:
Given the fatory pattern like `factory/kujira1643jxg8wasy5cfcn7xm8rd742yeazcksqlg4d7/umnta`,I utilized `substringAfter('/')` to extract relevant part